src/set_theory/game/basic.lean

```diff
@@ -590,16 +590,17 @@ noncomputable instance : has_inv pgame :=
 
 noncomputable instance : has_div pgame := ⟨λ x y, x * y⁻¹⟩
 
-theorem inv_eq_of_equiv_zero {x : pgame} (h : x ≈ 0) : x⁻¹ = 0 := if_pos h
+theorem inv_eq_of_equiv_zero {x : pgame} (h : x ≈ 0) : x⁻¹ = 0 :=
+by { classical, exact if_pos h }
```

Comment:
I'm a bit surprised this is needed. Doesn't the decidability instance get unified because we're directly giving the proof term?

If not, can you try whether the following lemma makes it work? Note the `{h : decidable c}`.
```
lemma if_pos' {c : Prop} {h : decidable c} (hc : c) {α : Sort u} {t e : α} : (ite c t e) = t :=
match h with
| (is_true  hc)  := rfl
| (is_false hnc) := absurd hc hnc
end
```

Reply:
This definitely doesn't work without this change.

Yes,
```
lemma if_pos' {c : Prop} {h : decidable c} (hc : c) {α : Sort u} {t e : α} : (ite c t e) = t :=
if_pos hc
```
and then proving this lemma with `if_pos' h` works fine.

Reply:
That's why we have quotient.mk' taking  a {setoid} argument instead of [setoid], right?

Reply:
I guess so, yes. I am not clear what the current lemma helps for given that you can't rewrite backwards with `if_pos`/`if_neg`/`dif_pos`/`dif_neg` anyway (and indeed I just checked that it's never used as such.

Given that this change somewhat reduces the diff of this PR and seems generally beneficial, I'd suggest just rolling along with it.

Reply:
`quotient.mk'` is a different beast because it's a definition intended for use in a term, not a lemma intended for use in a tactic.

> Given that this change somewhat reduces the diff of this PR and seems generally beneficial, I'd suggest just rolling along with it.

I've already set bors on the dependent PR (which contains the bulk of the fixes), since I want to ease the porting effort for `data.finsupp.defs` and similar. I'm not opposed to adding this new lemma, but I don't want to make this PR any harder to forward-port than it already will be.